hypothesis-python/src/hypothesis/core.py

```diff
@@ -536,9 +555,12 @@ def test(*args, **kwargs):
                 return result
 
         def run(data):
+            # Set up dynamic context needed by a single test run.
             with local_settings(self.settings):
```

Comment:
I wonder if we still need this. We no longer actually support nesting given, right?

(This is thinking out loud I'm not expecting you to do anything with it in this PR)

Reply:
given definitely refuses to wrap itself.

Reply:
```python
# Banned!
@given(x=...)
@given(y=...)
def test(x, y): ...

# Allowed!
@given(x=...)
def test(x):
    @given(y=...)
    def inner(y): ...
```

I think we want to continue allowing the latter form, so I'd leave this as-is.

Reply:
I definitely don't think we *want* to continue allowing the latter form, but I guess you're right that we've not currently deprecated it.